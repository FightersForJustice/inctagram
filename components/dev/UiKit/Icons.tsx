import { PageWrapper } from 'components/PageWrapper/PageWrapper'
import style from '@/@ui/ui-kit/Icon/IconsComponent.module.scss'
import { Icons } from '@/@ui/ui-kit/Icon/IconsComponent'

export const getStaticProps = async () => {
  return {
    props: {},
  }
}

const IconsDev = () => {
  return (
    <PageWrapper>
      <div className={style.kitContainer}>

        <div style={{ display: 'flex', textAlign: 'center', color: 'white' }}>
        <div>
          <h2>Default</h2>
          <Icons.Home className={style.SideBar__Item} />
          <Icons.Create className={style.SideBar__Item} />
          <Icons.Profile className={style.SideBar__Item} />
          <Icons.Messenger className={style.SideBar__Item} />
          <Icons.Search className={style.SideBar__Item} />
          <Icons.Statistics className={style.SideBar__Item} />
          <Icons.Logout className={style.SideBar__Item} />
        </div>
        <div>
          <h2>Active</h2>
          <Icons.Home className={style.SideBar__Item} isActive />
          <Icons.Create className={style.SideBar__Item} />
          <Icons.Profile className={style.SideBar__Item} />
          <Icons.Messenger className={style.SideBar__Item} />
          <Icons.Search className={style.SideBar__Item} />
          <Icons.Statistics className={style.SideBar__Item} />
          <Icons.Logout className={style.SideBar__Item} />
        </div>
        <div>
          <h2>Disabled</h2>
          <Icons.Home className={style.SideBar__Item} isDisabled />
          <Icons.Create className={style.SideBar__Item} />
          <Icons.Profile className={style.SideBar__Item} />
          <Icons.Messenger className={style.SideBar__Item} />
          <Icons.Search className={style.SideBar__Item} />
          <Icons.Statistics className={style.SideBar__Item} />
          <Icons.Logout className={style.SideBar__Item} />
        </div>
      </div>

      </div>
    </PageWrapper>
  )
}

export default IconsDev
