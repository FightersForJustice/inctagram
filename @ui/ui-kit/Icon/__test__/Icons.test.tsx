import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Icons from '../IconsComponent'
import style from '../IconsComponent.module.scss'

const iconsSubcomponent: Array<string> = ['Home', 'Create', 'Profile', 'Messenger', 'Search', 'Statistics', 'Favorites', 'Logout']
const disabledProps = [true, false]

describe('render snapshot correctly \t', () => {
  afterEach(cleanup)

  iconsSubcomponent.map((Subcomponent: string) => {
    disabledProps.map((isDisabled) => {
      it(`Subcomponent: ${Subcomponent} \t
      disabled: ${isDisabled} \t
      `, () => {
        const Icon = Icons[Subcomponent as keyof typeof Icons]
        const props = { isDisabled: isDisabled }
        const { asFragment } = render(<Icon {...props} />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})

describe('Icons', () => {
  afterEach(cleanup)
  it('renders correctly', () => {
    const { getByText } = render(<Icons.Home />)
    
    const IconElement = getByText('Home')
    expect(IconElement).toBeInTheDocument()
  })

  it('custom text renders correctly', () => {
    const { getByText } = render(<Icons.Home customText='test' />)
    
    const IconElement = getByText('test')
    expect(IconElement).toBeInTheDocument()
  }) 

  it('custom class renders correctly', () => {
    const { getByText } = render(<Icons.Home customClass='test' />)
    
    const IconElement = getByText('Home')
    expect(IconElement.closest('a')).toHaveClass('test')
  }) 

  it('redirects to the specified url', () => {
    const { getByText } = render(<Icons.Home url='test'  />)

    const IconElement = getByText('Home')
    expect(IconElement.closest('a')).toHaveAttribute('href', 'test')
  })

  it('link doesnt work when disabled', () => {
    const { getByText } = render(<Icons.Home url='test' isDisabled={true} />)

    const IconElement = getByText('Home')
    expect(IconElement.closest('a')).toBeFalsy()
  })

  it('have disabled class when disabled', () => {
    const { getByText } = render(<Icons.Home isDisabled={true} />)

    const IconElement = getByText('Home')
    expect(IconElement.closest('div')).toHaveClass(style.IconWrapperDisabled)
  })

  it('have active class when active', () => {
    const { getByText } = render(<Icons.Home isActive={true} />)

    const IconElement = getByText('Home')
    expect(IconElement.closest('a')).toHaveClass(style.IconWrapperActive)
  })

  it('link doesnt work when active and disabled', () => {
    const { getByText } = render(<Icons.Home isActive={true} isDisabled={true} />)

    const IconElement = getByText('Home')
    expect(IconElement.closest('a')).toBeFalsy()
  })

  it('have only disabled class when active and disabled', () => {
    const { getByText } = render(<Icons.Home isActive={true} isDisabled={true} />)

    const IconElement = getByText('Home')
    expect(IconElement.closest('div')).toHaveClass(style.IconWrapperDisabled)
  })
})
