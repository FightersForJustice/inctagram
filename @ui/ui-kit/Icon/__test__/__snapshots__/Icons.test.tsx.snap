// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot correctly 	 Subcomponent: Create 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Create
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Create 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Create
    </span>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Favorites 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Favorites
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Favorites 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Favorites
    </span>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Home 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Home
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Home 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Home
    </span>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Logout 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Logout
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Logout 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Logout
    </span>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Messenger 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Messenger
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Messenger 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Messenger
    </span>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Profile 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Profile
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Profile 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Profile
    </span>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Search 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Search
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Search 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Search
    </span>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Statistics 	
      disabled: false 	
       1`] = `
<DocumentFragment>
  <a
    class="IconWrapper"
    href=""
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Statistics
    </span>
  </a>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 Subcomponent: Statistics 	
      disabled: true 	
       1`] = `
<DocumentFragment>
  <div
    class="IconWrapper IconWrapperDisabled"
  >
    <div
      class="Icon"
    >
      <svg />
    </div>
    <span
      class="Text"
    >
      Statistics
    </span>
  </div>
</DocumentFragment>
`;
